docs/docs/reference-react-component.md

```diff
@@ -0,0 +1,261 @@
+---
+id: react-component
+title: React.Component
+layout: docs
+category: Reference
+permalink: docs/react-component.html
+---
+
+[Components](/react/docs/components-and-props.html) let you split the UI into independent, reusable pieces, and think about each piece in isolation.
+
+## Overview
+
+`React.Component` is an abstract base class, so it rarely makes sense to refer to `React.Component` directly. Instead, you will typically subclass it, and define at least a [`render()`](#render) method.
+
+Normally you would define a React component as a plain [JavaScript class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes):
+
+```javascript
+class Greeting extends React.Component {
+  render() {
+    return <h1>Hello, {this.props.name}</h1>;
+  }
+}
+```
+
+If you don't use ES6 yet, you may use the `React.createClass` helper instead. Take a look at [Using React without ES6](/react/docs/react-without-es6.html) to learn more.
+
+### The Component Lifecycle
+
+Each component has several "lifecycle methods" that you can override to run code at particular times in the process. Methods prefixed with **`will`** are called right before something happens, and methods prefixed with **`did`** are called right after something happens.
+
+#### Mounting
+
+These methods are called when an instance of a component is being created and inserted into the DOM:
+
+- [`constructor(props)`](#constructor)
+- [`componentWillMount()`](#componentwillmount)
+- [`render()`](#render)
+- [`componentDidMount()`](#componentdidmount)
+
+#### Updating
+
+An update can be caused by changes to props or state. These methods are called when a component is being re-rendered:
+
+- [`componentWillReceiveProps(nextProps)`](#componentwillreceivepropsnextprops)
+- [`shouldComponentUpdate(nextProps, nextState)`](#shouldcomponentupdatenextprops-nextstate)
+- [`componentWillUpdate(nextProps, nextState)`](#componentwillupdatenextprops-nextstate)
+- [`render()`](#render)
+- [`componentDidUpdate(prevProps, prevState)`](#componentdidupdateprevprops-prevstate) is invoked after an update and can be used to operate on the DOM after the component has been updated or to fetch new data in response to prop changes.
+
+#### Unmounting
+
+This method is called when a component is being removed from the DOM:
+
+- [`componentWillUnmount()`](#componentwillunmount)
+
+#### Other APIs
```

Comment:
Use a level 3 heading here as these APIs don't belond under The Component Lifecycle.